nemoguardrails/llm/providers/providers.py

```diff
@@ -22,22 +22,17 @@
 """
 
 import asyncio
+import importlib
 import logging
 import warnings
-from importlib.metadata import version
-from typing import Any, Dict, List, Optional, Type
+from functools import lru_cache
+from typing import Dict, List, Type
 
-from langchain.callbacks.manager import (
-    AsyncCallbackManagerForLLMRun,
-    CallbackManagerForLLMRun,
-)
-from langchain.schema.output import GenerationChunk
+from langchain.chat_models.base import BaseChatModel
 from langchain_community import llms
-from langchain_community.llms import HuggingFacePipeline
+from langchain_community.chat_models import _module_lookup
```

Comment:
add tests to ensure latest lc version still supports this (latest-test workflow will catch it)